include/seqan3/io/detail/misc_input.hpp

```diff
@@ -97,7 +98,7 @@ inline auto make_secondary_istream(std::basic_istream<char_t> & primary_stream,
     std::string const extension = filename.extension().string();
 
     // set return value appropriately
-    if (starts_with(magic_number, std::array{'\x1f', '\x8b', '\x08'})) // GZIP
```

Comment:
A few lines above the header is extracted, but only 4 bytes. You need to decide and hardcode how much you want to extract, but you need to make sure that you put everything back and also take into account that being at end before you read to the end might be valid, e.g. you may be at end, before you read the end of a bgzf magic header, but does that mean it is invalid or not compressed? What would
```
> A
C
```
(a valid fasta file) look compressed in Gzip? Is that smaller than the bgzf magic header?

Reply:
I know. But I haven‘t changed anything in the code except replacing the hardcoded byte arrays with the proper magic_header. BGZF is not added here yet. 
Regarding the end of file, the Bgzf header is always written no matter how small the input data. It then follows the compressed block in this case the fasta file followed by a Bgzf block footer and finally (after the last Bgzf block) the end of file marker. I guess in a valid gzip or bzip compressed file you will always have a valid header. Of course for small input sizes the compressed file could be bigger than the plain text files, but I guess this is of no concern.

Reply:
> I know. But I haven‘t changed anything in the code except replacing the hardcoded byte arrays with the proper magic_header. BGZF is not added here yet.

Ah, ok. Well it's something to easily forget, because it might work, even if you don't change it. Please make a big TODO (in code or in your mind :wink: ) to not forget this!

> Regarding the end of file, the Bgzf header is always written no matter how small the input data. 

I wasn't referring to this. Anyway, let's discuss it when the code is actually changed.

Reply:
Ah, I now understand what you mean. To test between gzip or bgzf you might be at the end before the magic header ends. Yes, this is important and taken care of.